.github/workflows/deploy-doc-site.yml

```diff
@@ -0,0 +1,82 @@
+name: Deploy documentation site
+
+on:
+  push:
+    tags: [ v* ]
+  pull_request:
+    # Run this workflow as a dry run when it's edited, so we can test it.
+    paths: .github/workflows/deploy-doc-site.yml
+
+jobs:
+  deploy:
+    runs-on: macOS-latest
```

Comment:
Outside of the scope of this PR, but I wonder if we really want to be using macOS-latest. Not hermetic.